site/content/en/development/design-docs/ipam.md

```diff
@@ -0,0 +1,38 @@
+---
+title: "Support node IPAM controller"
+linkTitle: "node IPAM controller"
+type: docs
+weight: 10
+description: >
+Support node IPAM controller in the out-of-tree provider Azure.
+---
+
+## Background
+
+The in-tree node IPAM controller would be moved to the GCE staging repo in the future, 
```

Comment:
The in-tree [Node IPAM controller](https://github.com/kubernetes/kubernetes/tree/master/pkg/controller/nodeipam) only supports a fixed node CIDR mask size for all nodes, while in multiple node pool (VMSS) scenarios, different mask sizes are required for different node pools. There is a GCE-specific cloud CIDR allocator for a similar scenario, but that is not exposed in cloud provider API and it is planned to be moved out-of-tree. Hence this docs proposes an out-of-tree node IPAM controller.

Reply:
actually the custom mask size is support in the in-tree node ipam controller by a kcm flag `node-cidr-mask-size`